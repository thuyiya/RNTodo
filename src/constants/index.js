const theme = {
    colors: {
        primary: '#0145a6',
        textColor: '#333333',
        seconderyTextColor: '#787878',
        backdropColor: '#000000',
        loadingIndicator: '#000000',
        background: '#f0f0f0',
        disbaleText: '#c9c9c9',
        white: '#ffffff',
        buttonText: '#ffffff',
    },
    opacity: {
        backdrop: 0.3
    },
    fontSizeMap: {
        h1: 24,
        h2: 22,
        h3: 20,
        h4: 18,
        h5: 16,
        h6: 14
    },
    fontWeightMap: {
        normal: '300',
        avarage: '400',
        lightBold: '500',
        midBold: '700',
        bold: '900',
    },
    spaceSize: 8,
}

const strings = {
    AUTHENTICATION_SCREEN: {
        DEVICE_CAPABILITY: 'No face or fingerprint scanner is available on the device.',
        PROCCED_TEXT: 'Set Authentication to Proceed',
        GO_TO_SETTINGS: 'Go to Settings',
        PROCEED_TO_AUTH: 'Proceed',
        GO_TO_SETTINGS_ALERT_TITLE: 'Authentication Setup',
        GO_TO_SETTINGS_ALERT_DESCRIPTION: 'Go to settings and setup authentication to proceed',
        APP_ACCESS: 'Access setup',
        APP_ACCESS_DESCRIPTION: 'You have to pass your authentication process to see todos',
    },
    HOME: {
        TITLE: 'TODO: '
    },
    EMPTYVIEW: {
        NO_DATA: "No Todo's Found"
    }
}

const actionTypes = {
    TODO: {
        ADD_ITEM: 'TODO_ADD_ITEM',
        CLEAN_ITEMS: 'CLEAN_ITEMS',
        REMOVE_ITEMS: 'REMOVE_ITEMS',
        SET_SELECTED_ITEM: 'SET_SELECTED_ITEM'
    }
}

export { theme, strings, actionTypes }