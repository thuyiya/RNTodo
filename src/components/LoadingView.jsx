import React from 'react';
import { View, StyleSheet, Dimensions, Text, ActivityIndicator } from 'react-native';
import { theme } from '../constants';
import { Typography } from './';

const { height, width } = Dimensions.get('screen');

const Loading = ({ enable, text }) => {

    if(!enable) {
        return null
    }

    return (<View style={styles.container}> 
            <View style={styles.backdrop}/>
            <ActivityIndicator size={"small"} color={theme.colors.loadingIndicator}/>
            <Typography style={styles.text}>{text}</Typography>
    </View>)
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        position: 'absolute',
    },
    backdrop: {
        position: 'absolute',
        backgroundColor: theme.colors.backdropColor,
        opacity: theme.opacity.backdrop,
        height,
        width,
        zIndex: 9999999,
    },
    text: {
        zIndex: 99999999
    }
})

export default Loading;