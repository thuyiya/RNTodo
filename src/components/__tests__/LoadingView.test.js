import React from "react";
import renderer from "react-test-renderer";
import {render, screen} from '@testing-library/react-native';
import LoadingView from "../LoadingView";
import { strings } from "../../constants";

describe("<LoadingView />", () => {
  it("has 2 children", () => {
    const tree = renderer.create(<LoadingView enable={true} />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it("no tree rendering for no load", () => {
    const tree = renderer.create(<LoadingView enable={false} />).toJSON();
    expect(tree).toBe(null);
  });

  it(`displays the loading test as Loading`, () => {
    render(<LoadingView enable={true} text="Loading" />);
    expect(screen.queryByText("Loading")).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<LoadingView enable={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
