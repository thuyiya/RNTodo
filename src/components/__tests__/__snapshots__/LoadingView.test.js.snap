// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingView /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "position": "absolute",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "height": 1334,
        "opacity": 0.3,
        "position": "absolute",
        "width": 750,
        "zIndex": 9999999,
      }
    }
  />
  <ActivityIndicator
    color="#000000"
    size="small"
  />
</View>
`;
